### ISSUE TITLE
🚀 Feature Request: Reporting incidents through API/CLI

### ISSUE THREAD

**Issue by tehtnaz** [OnlineOrNot/onlineornot#47]
### Describe the solution

✓ I've read the roadmap

It would be nice to be able to report incidents using the API/CLI. This would be useful, for example, when having scheduled maintenance because then it would be possible to easily run a script to post an incident without needing to open a browser (eg. running the script server-side).

So, essentially, here's my wishlist

- [ ] Add a new endpoint, for example, /incidents (and the corresponding scope to API tokens to allow requests to be made)
- [ ] Documentation for /incidents
- [ ] (Maybe) CLI implementation of /incidents

FYI, I'm willing to contribute to the CLI. I'll submit a draft PR with what the new commands could look like.

**Comment by rozenmd** [comment]
Good point, I'll update the API to have an OpenAPI spec for the existing public endpoints, then add new endpoints for status pages.

**Comment by rozenmd** [comment]
OpenAPI spec is now generating docs: https://developers.onlineornot.com/

Still need to:
- [x] make a repo to publish the openapi spec itself
- [ ] make new endpoints
